api/src/common_items.js

```diff
@@ -0,0 +1,216 @@
+}
+
+// (NOT USED)
+function clean_input_encodeurl(string) {
+    //string = string.replaceAll(/[']/ig, '%27');
+    string = encodeURIComponent(string);
+    return string;
+}
+
+// (NOT USED)
+function unescape_object(obj) {
+    for (let key in obj) {
+        // javascript is a perfect language with no flaws
+        if (Object.prototype.hasOwnProperty.call(obj, key) && obj[key] && typeof(obj[key]) === "string") {
+            //obj[key] = obj[key].replaceAll(/[%27]/ig, "'");
+            obj[key] = decodeURIComponent(obj[key]);
+        }
+    }
+
+    return obj;
+}
+// -------------------------------------------------
+
+// -------------- fiscal year globals -------------
+
+/** CHANGE BELOW ANNUALLY **/
+const current_fiscal_year = "2021-2022";
+/** CHANGE ABOVE ANNUALLY **/
+
+const first_fiscal_year = "2015-2016";
+const yearStart = parseInt(first_fiscal_year.substring(0,4));
+const yearEnd = parseInt(current_fiscal_year.substring(0,4));
+const fiscal_year_list = [];
+for (let year = yearEnd; year >= yearStart; year--) {
+    fiscal_year_list.push(`${year}-${year+1}`);
+}
+// -------------------------------------------------
+
+// ------------ financial committee lut ------------
+// this is the best solution I can come up with
+//  with out a database schema migration. Here is the problem:
+//  the committee column in the database is an enum with some values,
+//  but the name of the committee does not match the enum value.
+//  Further, the name of the committee is not a http safe thing to put in a URL.
+//  Therefore, this is a lookup table of sorts to cross reference these
+//  three different names for THE SAME COMMITEE
+// format = { http-name: [ db enum, committee name ] }
+const committee_lut =
+{
+    "general":["General IEEE", "General IEEE"],
+    "aerial":["Aerial Robotics", "Aerial Robotics"],
+    "csociety":["Computer Society", "Computer Society"],
+    "embs":["EMBS", "EMBS"],
+    "mtt-s":["MTT-S", "MTT-S"],
+    "racing":["Racing", "Racing"],
+    "rov":["ROV", "ROV"],
+    "soga":["SOGA", "SOGA"],
+};
+// mini-LUT for db enum : committee name
+const committee_name_swap =
+{
+    "General IEEE":"General IEEE",
+    "Aerial Robotics":"Aerial Robotics",
+    "Computer Society":"Computer Society",
+    "EMBS":"EMBS",
+    "MTT-S":"MTT-S",
+    "Racing":"Racing",
+    "ROV":"ROV",
+    "SOGA":"SOGA",
+};
+// -------------------------------------------------
+
+// ------------- dues committee lut ----------------
+// List of all committees a member can be part of
+const dues_committees =
+[
+    "Aerial Robotics",
+    "Computer Society",
+    "EMBS",
+    "Growth & Engagement",
+    "MTT-S",
+    "Industrial Relations",
+    "Learning",
+    "Racing",
+    "ROV",
+    "Social",
+    "Software Saturdays"
+];
+// -------------------------------------------------
+
+// ----------------- dues amount -------------------
+// Current annual local dues
+
+/** CHANGE BELOW ANNUALLY **/
+const dues_amount = 15.0;
```

Comment:
Haven't looked at where all this is used, but there were times (like 2020-21) when there were separate oneline vs in person dues. Also, how do we handle people paying IEEE international dues, since we passed the amendment to count people as having paid local if they pay international.

Reply:
That is a good catch, I do not account for that at the moment

Reply:
Yah. I went kinda rudimentary in v1 (not sure how much you read / kept) but it counted someone as paying dues as long as they paid more than 0. We also had a few people (for whatever reason) pay Ryan like $14, which wasn't quite right

Reply:
Not a huge deal nor worth blocking merging, so maybe something for over the summer. Especially since new dues aren't really being collected now.